cpp/src/io/json/read_json.cu

```diff
@@ -270,6 +274,15 @@ table_with_metadata read_batch(host_span<std::unique_ptr<datasource>> sources,
   return device_parse_nested_json(buffer, reader_opts, stream, mr);
 }
 
+size_t get_batch_size_upper_bound()
+{
+  auto const batch_size_str = std::getenv("LIBCUDF_JSON_BATCH_SIZE");
+  size_t const batch_size   = batch_size_str != nullptr ? std::atol(batch_size_str) : 0L;
+  return (batch_size > 0 && batch_size < std::numeric_limits<int32_t>::max())
```

Comment:
Oh this will either trigger compiler warning/error or `size_t` will overflow.
```suggestion
  int64_t const batch_size   = batch_size_str != nullptr ? std::atol(batch_size_str) : 0L;
  return (batch_size > 0 && batch_size < std::numeric_limits<int32_t>::max())
```

Reply:
And by changing `batch_size` type, we also need to change the return type.

Reply:
Fixed!